test/CMakeLists.txt

```diff
@@ -92,6 +97,23 @@ foreach(file ${sources})
                     set(TestEthArgs -t ${TestSuitePathFixed}/${TestCase} -- ${TESTETH_ARGS})
                     add_test(NAME ${TestSuitePathFixed}/${TestCase} WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/test COMMAND testeth ${TestEthArgs})
                     set_tests_properties(${TestSuitePathFixed}/${TestCase} PROPERTIES TIMEOUT ${timeout})
+
+                    list(LENGTH allTestsPart1 allTestsPart1Length)
+                    list(LENGTH allTestsPart2 allTestsPart2Length)
+                    list(LENGTH allTestsPart3 allTestsPart3Length)
+                    list(LENGTH allTestsPart4 allTestsPart4Length)
+                    list(LENGTH allTestsPart5 allTestsPart5Length)
+                    if(allTestsPart1Length LESS 100)
+                        list(APPEND allTestsPart1 "${TestSuitePathFixed}/${TestCase} ")
+                    elseif(allTestsPart2Length LESS 100)
+                        list(APPEND allTestsPart2 "${TestSuitePathFixed}/${TestCase} ")
+                    elseif(allTestsPart3Length LESS 100)
+                        list(APPEND allTestsPart3 "${TestSuitePathFixed}/${TestCase} ")
+                    elseif(allTestsPart4Length LESS 100)
+                        list(APPEND allTestsPart4 "${TestSuitePathFixed}/${TestCase} ")
+                    elseif(allTestsPart5Length LESS 100)
```

Comment:
This section looks horrible and will not work if you have more than 500 test suites.
Do it as @gumb0 proposed by creating an array of names. Here you have an example how to do it:
https://gist.github.com/chfast/d738c74d710d29bcb05de42c948be41a.

Reply:
Thanks! @chfast, your cmake mastery is awesome!